osf/models/node.py

```diff
@@ -2787,30 +2790,33 @@ def remove_node(self, auth, date=None):
         :param datetime date: `datetime.datetime` or `None`
         """
         # TODO: rename "date" param - it's shadowing a global
-        if not self.can_edit(auth):
+        hierarchy = Node.objects.get_children(self, active=True, include_root=True)
+        is_admin = Contributor.objects.filter(node=OuterRef('pk'), admin=True, user=auth.user)
+        if len(hierarchy.annotate(has_admin_perm=Exists(is_admin)).filter(has_admin_perm=False)):
             raise PermissionsError(
-                '{0!r} does not have permission to modify this {1}'.format(auth.user, self.category or 'node')
+                '{0!r} does not have permission to modify this {1}, or a component in its hierarchy.'.format(auth.user, self.category or 'node')
             )
 
-        if Node.objects.get_children(self, active=True).exists():
-            raise NodeStateError('Any child components must be deleted prior to deleting this project.')
+        has_public = hierarchy.filter(is_public=True).exists()
 
         # After delete callback
-        remove_addons(auth, [self])
-
-        log_date = date or timezone.now()
+        remove_addons(auth, hierarchy)
 
         Comment = apps.get_model('osf.Comment')
-        Comment.objects.filter(node=self).update(root_target=None)
+        Comment.objects.filter(node__id__in=hierarchy).update(root_target=None)
 
-        # Add log to parent
-        self.add_remove_node_log(auth=auth, date=log_date)
+        log_date = date or timezone.now()
+        for node in hierarchy:
+            # Add log to parents
+            node.is_deleted = True
+            node.deleted_date = date
+            self.add_remove_node_log(auth=auth, date=log_date)
+            project_signals.node_deleted.send(node)
 
-        self.is_deleted = True
-        self.deleted_date = date
-        self.save()
+        bulk_update(hierarchy, update_fields=['is_deleted', 'deleted_date'])
```

Comment:
I tried calling hierarchy.update(is_deleted=True, deleted_date=date), but then the hierarchy would have needed to be reloaded from the db before being passed into bulk_update_search.  Deleted_changes wouldn't be reflected otherwise.